components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -27,6 +30,51 @@ fn json_array(args: &[&Option<Json>]) -> Result<Option<Json>> {
     )))
 }
 
+fn json_object_validator(expr: &tipb::Expr) -> Result<()> {
+    for chunk in expr.get_children().chunks(2) {
+        if chunk.len() == 1 {
```

Comment:
Looks like we can check it as the first step, instead of check it in each loop iteration.